### PR TITLE
Removed core dependency

### PR DESCRIPTION
**Pull request by JakeGinnivan**

### CODE REVIEW

File: src/GitTools.IssueTrackers/AuthSettings.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        {
+            if (string.IsNullOrEmpty(token)) throw new ArgumentNullException("token");
+            Token = token;
+            IsTokenAuthentication = true;
+        }
+
+        public bool IsEmpty { get; private set; }
+
+        public string Token { get; private set; }
+        public bool IsTokenAuthentication { get; private set; }
```
Comment (GeertvanHorrik) [inline src/GitTools.IssueTrackers/AuthSettings.cs:32]: I always find duplicate values very cumbersome. Use calculated properties or extension methods instead?

File: src/GitTools.IssueTrackers/AuthSettings.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    using JetBrains.Annotations;
+
+    public class AuthSettings
+    {
+        public AuthSettings()
+        {
+            IsEmpty = true;
+        }
+
+        public AuthSettings([NotNull] string username, [NotNull] string password)
```
Comment (GeertvanHorrik) [inline src/GitTools.IssueTrackers/AuthSettings.cs:13]: Having different auth contexts is a bad idea. Isn't that the whole point of GitTools.Core?

Comment (JakeGinnivan) [inline src/GitTools.IssueTrackers/AuthSettings.cs:13]: Its 3 properties. And issue tracker auth is different to git auth. You don't authenticate to your issue tracker with SSH. We also drop the libgit2sharp dependency which is not needed in this package :)

Trying to overreuse is just as bad. GitTools.Core for me has some clear goals, which is dynamic repos, repo normalisation and other complex git related things.

If we find we need some common data structures then maybe we have GitTools.Common down the track.

File: src/GitTools.IssueTrackers/AuthSettings.cs (hunk 1 above)
Comment (JakeGinnivan) [inline src/GitTools.IssueTrackers/AuthSettings.cs:32]: This class is immutable at the moment, we have no need to recalculate and these can be set straight off the ctor.

**Pull request merged**